jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -14,4 +21,69 @@ public class JdbcTemplate {
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public void update(final String sql, final Object...args) {
+        try (final Connection conn = dataSource.getConnection();
+             final PreparedStatement pstmt = conn.prepareStatement(sql)
+        ) {
+            setValues(pstmt, args);
+            log.debug("query : {}", sql);
+
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            log.error("JdbcTemplate {} failed. SQL: {}, args: {}", "update", sql, args, e);
+            throw new DataAccessException("JdbcTemplate query failed", e);
+        }
+    }
+
+    public <T> T queryForObject(final String sql, RowMapper<T> rowMapper, final Object...args) {
+        try (final Connection conn = dataSource.getConnection();
+             final PreparedStatement pstmt = conn.prepareStatement(sql)
+        ) {
+            setValues(pstmt, args);
+            log.debug("query : {}", sql);
+
+            try (ResultSet rs = pstmt.executeQuery()) {
+                if (rs.next()) {
+                    return rowMapper.mapRow(rs);
+                }
+            }
+            return null;
```

Comment:
매핑할 객체가 없다면 null을 반환해주고, 외부에서 옵셔널로 가마서 받고 있는 것 같아요. 
해당 메서드를 옵셔널로 반환시키는 방법도 있을 것 같아요! 

이렇게 구현하신 이유가 궁금합니다~

```suggestion
    public <T> Optional<T> queryForObject(final String sql, RowMapper<T> rowMapper, final Object...args) {
        try (final Connection conn = dataSource.getConnection();
             final PreparedStatement pstmt = conn.prepareStatement(sql)
        ) {
            setValues(pstmt, args);
            log.debug("query : {}", sql);

            try (ResultSet rs = pstmt.executeQuery()) {
                if (rs.next()) {
                    return  Optional.of(rowMapper.mapRow(rs));
                }
            }
            return Optional.empty();
```

Reply:
JdbcTemplate의 책임은 SQL을 실행하고 결과를 매핑하는 것이라고 생각했고, 결과가 없을 때 어떻게 처리할 것인지에 대한 문제는 UserDao의 책임이라고 생각해서 이와 같이 구현했습니다!

또한, JdbcTemplate에서 옵셔널 반환을 강제하게 되면, 데이터가 반드시 있어야 하는 상황에서 데이터가 조회되지 않는 것과 데이터 자체가 정상적으로 존재하지 않는 상황을 구분할 수 없을 것이라고 판단했어요